src/query/atomWithQuery.ts

```diff
@@ -60,7 +61,10 @@ export function atomWithQuery<
             }
             action.initializer(getQueryClient(get, set))
           } else if (action.type === 'data') {
-            set(dataAtom, action.data)
+            const data = get(dataAtom)
+            if (!data || !equalityFn(data, action.data)) {
```

Comment:
```suggestion
            if (data === null || !equalityFn(data, action.data)) {
```
Let's make it strict here. (tbh, i'm not certain if `null` in L37 is valid to mean uninitialized. Can `TData` ever be `null`?)